bokehjs/src/lib/models/glyphs/webgl/webgl_utils.ts

```diff
@@ -0,0 +1,90 @@
+import {LineJoin} from "core/enums"
+import {Uniform} from "core/uniforms"
+import {color2rgba} from "core/util/color"
+import {uint32} from "core/types"
+import {HatchPattern} from "core/property_mixins"
+import {hatch_aliases} from "core/visuals/patterns"
+
+
+// WebGL shaders use integers for caps, joins and hatching.
+export const cap_lookup = {butt: 0, round: 1, square: 2}
+
+export const join_lookup = {miter: 0, round: 1, bevel: 2}
+
+const hatch_pattern_lookup: {[key: string]: number} = {
+  blank: 0,
+  dot: 1,
+  ring: 2,
+  horizontal_line: 3,
+  vertical_line: 4,
+  cross: 5,
+  horizontal_dash: 6,
+  vertical_dash: 7,
+  spiral: 8,
+  right_diagonal_line: 9,
+  left_diagonal_line: 10,
+  diagonal_cross: 11,
+  right_diagonal_dash: 12,
+  left_diagonal_dash: 13,
+  horizontal_wave: 14,
+  vertical_wave: 15,
+  criss_cross: 16,
+}
+
+
+function hatch_pattern_to_index(pattern: HatchPattern): number {
+  return hatch_pattern_lookup[hatch_aliases[pattern] ?? pattern] | 0
+}
+
+export function color_to_uint8_array(color_prop: Uniform<uint32>, alpha_prop: Uniform<number>): Uint8Array {
+  const ncolors: number = Math.max(color_prop.length, alpha_prop.length)
+  const rgba: Uint8Array = new Uint8Array(4*ncolors)
```

Comment:
There is no need to specify types twice (old Java-style) within function body. Take advantage of type inference.

Reply:
Done.